### PR TITLE
geolocation: add result destination for LocationOffsets

### LINKED ISSUE #3468: geolocation: add result destination to GeolocationUser (onchain + CLI + SDK)
Parent: #3430

## Summary

Add `result_destination_ip` and `result_destination_port` fields to the `GeolocationUser` onchain account, with a new instruction to set/clear them, CLI support, and Go SDK deserialization.

## Onchain changes

**`GeolocationUser` struct** (`smartcontract/programs/doublezero-geolocation/src/state/geolocation_user.rs`):
- Switch from `BorshDeserialize` to `BorshDeserializeIncremental`
- Append two fields after `targets`:
  - `result_destination_ip: Ipv4Addr` — `#[incremental(default = Ipv4Addr::UNSPECIFIED)]` (4 bytes)
  - `result_destination_port: u16` — `#[incremental(default = 0)]` (2 bytes)
- `0.0.0.0:0` means "not set" (current behavior preserved)

**New instruction: `SetResultDestination`**
- Args: `ip: Ipv4Addr`, `port: u16`
- Accounts: `user_account` (writable), `payer_account` (signer, must be owner)
- Validates IP is public (reuse `validate_public_ip`) when setting; allows `0.0.0.0:0` to clear
- Bumps `target_update_count` on all referenced probes so agents re-discover the change

**Backwards compatibility:**
- Follows the same pattern as `GeoProbe.target_update_count` (`BorshDeserializeIncremental` + `#[incremental(default)]`)
- Existing accounts without these bytes deserialize with defaults
- Add a backwards-compat test (serialize without new fields, verify deserialization defaults)

## CLI changes

**`smartcontract/cli/src/geolocation/user/`**:
- New command: `dz-cli geolocation user set-result-destination --user <code> --ip <addr> --port <port>`
- To clear: `dz-cli geolocation user set-result-destination --user <code> --clear`
- Display result destination in `dz-cli geolocation user get` output

## Go SDK changes

**`sdk/geolocation/go/state.go`**:
- Add `ResultDestinationIP [4]byte` and `ResultDestinationPort uint16` to `GeolocationUser` struct
- Deserialize with "try decode, default on error" pattern (same as `GeoProbe.TargetUpdateCount`)
- Add backwards-compat test (truncated data defaults to zero values)

## Testing

- Rust unit tests: serialization roundtrip with new fields, backwards-compat without new fields
- Rust integration tests: `SetResultDestination` success, unauthorized signer, invalid IP, clear
- Go SDK: deserialization with and without new fields
- Fixture regeneration (`make generate-fixtures`)

### PR DESCRIPTION
**Pull request by ben-dz**
Resolves: #3468

## Summary of Changes
- Rust SDK `SetResultDestinationCommand` that builds the onchain transaction with user PDA + probe account metas
- CLI `user set-result-destination --destination <host:port>` (or `--clear`) with client-side validation of IP routability and domain format before transaction submission
- Display `result_destination` in `user get` output (shows `"none"` when unset)
- Go SDK `ResultDestination string` field on `GeolocationUser` with backwards-compatible deserialization (old accounts default to `""`)
- Subcommand registration in the geolocation CLI binary

Stacked on #3501 which adds the onchain `SetResultDestination` instruction and `result_destination` field.

## Diff Breakdown
| Category    | Files | Lines (+/-)  | Net   |
|-------------|-------|--------------|-------|
| Core logic  |     3 | +425 / -8    | +417  |
| Scaffolding |     6 | +13 / -0     | +13   |
| Tests       |     1 | +49 / -2     | +47   |

Core logic is the CLI command (290 lines including validation + tests), Rust SDK command (118 lines), and Go SDK deserialization. Scaffolding is trait wiring and subcommand registration.

<details>
<summary>Key files (click to expand)</summary>

- `smartcontract/cli/src/geolocation/user/set_result_destination.rs` — CLI command with clap args, client-side `host:port` validation (IP routability + domain format per RFC 1035), mock tests for set/clear/invalid inputs
- `smartcontract/sdk/rs/src/geolocation/geolocation_user/set_result_destination.rs` — Rust SDK command: validates code, derives user PDA, passes probe account metas to `execute_transaction`
- `sdk/geolocation/go/state.go` — `ResultDestination string` field, serialize/deserialize with try-decode-default for backwards compat
- `sdk/geolocation/go/state_test.go` — round-trip and backwards-compat tests (truncated data defaults to empty)
- `smartcontract/cli/src/geoclicommand.rs` — `set_result_destination` trait method + impl

</details>

## Testing Verification
- CLI mock tests: set destination, clear, missing destination flag, 9 invalid destination cases (private IPs, bad ports, malformed domains)
- Go SDK: round-trip test with non-empty destination, empty-targets test, backwards-compat test (truncated data)
- All tests pass: 289 CLI, 165 SDK, 82 geolocation, Go SDK green

### CODE REVIEW

File: smartcontract/sdk/rs/src/geolocation/geolocation_user/set_result_destination.rs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+impl SetResultDestinationCommand {
+    pub fn execute(&self, client: &dyn GeolocationClient) -> eyre::Result<Signature> {
+        validate_code_length(&self.code)?;
+        let code =
+            validate_account_code(&self.code).map_err(|err| eyre::eyre!("invalid code: {err}"))?;
+
+        let program_id = client.get_program_id();
+        let (user_pda, _) = pda::get_geolocation_user_pda(&program_id, &code);
+
+        let mut accounts = vec![AccountMeta::new(user_pda, false)];
```
Comment (juan-malbeclabs) [inline smartcontract/sdk/rs/src/geolocation/geolocation_user/set_result_destination.rs:27]: `execute_transaction` always appends [payer, system_program] to the account list. So the current code produces this wire order:

[user_pda, probe_pk_0, probe_pk_1, ..., payer, system_program]

But the onchain processor (process_set_result_destination) expects:
https://github.com/malbeclabs/doublezero/blob/de9b9858c9f8c565fcca961d9c3b0bfd5f9acdf6/smartcontract/programs/doublezero-geolocation/src/processors/geolocation_user/set_result_destination.rs#L88
  accounts[0] = user_account
  accounts[1] = payer_account      ← signer, but gets probe_pk_0 instead
  accounts[2] = system_program     ← gets probe_pk_1 or payer
  accounts[3+] = probe_accounts

The unit tests miss this because they use a mock client that accepts whatever accounts are passed. The bug will only surface at runtime with a MissingRequiredSignature or IllegalOwner error from the program.

**Review by juan-malbeclabs — CHANGES_REQUESTED** [review]
Please, review Account Order

File: CHANGELOG.md [hunk 2]
```diff
@@ -62,6 +56,7 @@ All notable changes to this project will be documented in this file.
   - Add `IsRetryableFunc` field to `RetryOptions` for configurable retry criteria in the Solana JSON-RPC client; add `"rate limited"` string match and RPC code `-32429` to the default implementation
 - Geolocation
   - Standardize CLI flag naming: probe mutation commands use `--probe` (was `--code`) accepting pubkey or code; rename `--signing-keypair` → `--signing-pubkey` and `--target-pk` → `--target-signing-pubkey`; add `--json-compact` to `get` commands
+  - Add optional result destination to `GeolocationUser` so LocationOffsets can be sent to an alternate endpoint instead of the target IP; supports both IP and domain destinations (e.g., `185.199.108.1:9000` or `results.example.com:9000`); includes `SetResultDestination` onchain instruction, CLI `user set-result-destination` command, and Go SDK deserialization (backwards-compatible with existing accounts)
```
Comment (juan-malbeclabs) [inline CHANGELOG.md]: The new entry is inside the ## [v0.17.0] block instead of ## Unreleased. Move it to the Unreleased section.

File: smartcontract/programs/doublezero-geolocation/src/processors/geolocation_user/set_result_destination.rs [hunk 3]
```diff
@@ -124,18 +127,16 @@ pub fn process_set_result_destination(
...
-    let remaining: Vec<&AccountInfo> = accounts_iter.collect();
-    if remaining.len() != unique_probes.len() {
+    if probe_accounts.len() != unique_probes.len() {
         msg!(
             "Expected {} probe accounts, got {}",
             unique_probes.len(),
-            remaining.len()
+            probe_accounts.len()
         );
         return Err(GeolocationError::TooManyReferencedProbes.into());
```
Comment (juan-malbeclabs) [inline smartcontract/programs/doublezero-geolocation/src/processors/geolocation_user/set_result_destination.rs]: TooManyReferencedProbes fires for both too-many and too-few probe accounts. A caller passing zero probes for a user with targets receives this error, which is misleading. Consider renaming to ProbeAccountCountMismatch.

**Review by juan-malbeclabs — APPROVED** [review]
LGTM.Two more comments: the first is to prevent the changelog from ending up incorrect, and the second is purely semantic

File: CHANGELOG.md (hunk 2 above)
Comment (ben-dz) [reply CHANGELOG.md]: Whoops.
